spec/support/metric_helper.rb

```diff
@@ -1,46 +1,54 @@
 module Spec
   module Support
     module MetricHelper
-      # given (enabled) capture_targets, compare with suggested queue entries
-      def metric_targets(expected_targets)
-        expected_targets.flat_map do |t|
-          # Storage is hourly only
-          # Non-storage historical is expecting 7 days back, plus partial day = 8
-          t.kind_of?(Storage) ? [[t, "hourly"]] : [[t, "realtime"]] + [[t, "historical"]] * 8
-        end
-      end
-
-      # @return [Array<Array<Object, String>>] List of object and interval names in miq queue
-      def queue_intervals(items = MiqQueue.where(:method_name => %w[perf_capture_hourly perf_capture_realtime perf_capture_historical]))
-        items.map do |q|
-          interval_name = q.method_name.sub("perf_capture_", "")
-          [Object.const_get(q.class_name).find(q.instance_id), interval_name]
-        end
-      end
-
       # method_name => {target => [timing1, timing2] }
       # for each capture type, what objects are submitted and what are their time frames
       # @return [Hash{String => Hash{Object => Array<Array>}} ]
       def queue_timings(items = MiqQueue.where(:method_name => %w[perf_capture_hourly perf_capture_realtime perf_capture_historical]))
         messages = {}
         items.each do |q|
-          obj = q.instance_id ? Object.const_get(q.class_name).find(q.instance_id) : q.class_name.constantize
+          klass = q.class_name.constantize.base_class.name
+          # exploding looks into the third argument which contains the list of object ids
+          ids = q.args[2]
+          ids = [q.instance_id] if ids.blank?
+          objs = ids.map { |id| "#{klass}:#{id}" }.sort
```

Comment:
This won't do a numerical sort if that's what you're going for?  Perhaps:

```suggestion
          objs = ids.sort.map { |id| "#{klass}:#{id}" }
```

Reply:
not sure which one we want for tests. But that change looks as good as any